.github/workflows/e2e-pr.yml

```diff
@@ -0,0 +1,185 @@
+name: E2E
```

Comment:
These two YML files are nearly identical; could we attempt to combine them into one and use a conditional to determine if the run is a pull request or a push?